apps/src/templates/teacherDashboard/EditSectionForm.jsx

```diff
@@ -177,7 +130,147 @@ export default connect(state => ({
   validAssignments: state.teacherSections.validAssignments,
   primaryAssignmentIds: state.teacherSections.primaryAssignmentIds,
   sections: state.teacherSections.sections,
-}))(EditSectionForm);
+  section: state.teacherSections.sectionBeingEdited,
+  isSaveInProgress: state.teacherSections.saveInProgress,
+}), {
+  editSectionProperties,
+  handleSave: finishEditingSection,
+  handleClose: cancelEditingSection,
+})(EditSectionForm);
+
+const FieldProps = {
+  value: PropTypes.any,
+  onChange: PropTypes.func.isRequired,
+  disabled: PropTypes.bool,
+};
+
+const SectionNameField = ({value, onChange, disabled}) => (
+  <div>
+    <FieldName>
+      {i18n.sectionName()}
+    </FieldName>
+    <FieldDescription>
+      {i18n.addSectionName()}
+    </FieldDescription>
+    <input
+      value={value}
+      onChange={event => onChange(event.target.value)}
+      style={style.sectionNameInput}
+      disabled={disabled}
+    />
+  </div>
+);
+SectionNameField.propTypes = FieldProps;
+
+const GradeField = ({value, onChange, validGrades, disabled}) => {
+  const gradeOptions = [""]
+    .concat(validGrades)
+    .map(grade => ({
+      value: grade,
+      text: grade === 'Other' ? 'Other/Mixed' : grade,
```

Comment:
We do get this from the server, stored as a plain string.  Better to update the canonical value on the server and migrate existing rows, or just customize display here?  As for localization, I wasn't sure whether that's called for here or not - we don't localize 'K', '1', '2', '3'... but they're probably _mostly_ universal?

Especially if we want to do this on the server, I'll argue for doing that later so that this change is isolated to the new section flow and doesn't affect the old table.